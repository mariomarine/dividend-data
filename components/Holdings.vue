<template>
	<div class="holdings">
		<h2>Holdings</h2>
		<div class="holdings-table">
			<div class="holdings-row">
				<span> Ticker </span>
				<span> Shares </span>
				<span> Price </span>
			</div>
			<Holding  v-for="stock in stocks" :key="stock.ticker">
				<span slot="ticker"> {{ stock.ticker }} </span>
				<span slot="shares"> {{ stock.shares }} </span>
				<span slot="price"> {{ stock.price }}</span>
			</Holding>
		</div>
	</div>
</template>

<script>
import Holding from '@/components/Holding.vue'

export default {
	components: {
		Holding
	},
	data() {
		return {
			ticker: 'SO',
			shares: 28.1,
			stocks: [
				{
					ticker: 'SO',
					shares: 28.1,
					price: '$84.87'
				},
				{
					ticker: 'DUK',
					shares: 14,
					price: '$114.91'
				},
				{
					ticker: 'NEE',
					shares: 15,
					price: '$11.91'
				}
			]
		}
	}
}
</script>

<style>
.holdings-table {
	display: grid;
	grid-template-rows: auto;
}
.holdings-row {
	display: grid;
	grid-template-columns: 1fr 1fr 1fr;
}
</style>

