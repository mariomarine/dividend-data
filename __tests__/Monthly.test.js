import { mount } from '@vue/test-utils';
import Holdings from '../components/Monthly.vue';

describe('Monthly.vue', () => {
  test('should mount for testing', () => {
    expect(1).toEqual(1);
  });
  test('should render content correctly', () => {
	const wrapper = mount(Holdings);
	expect(wrapper.vm.$el).toMatchSnapshot();
  });
});

