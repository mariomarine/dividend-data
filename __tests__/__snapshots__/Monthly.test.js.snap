// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monthly.vue should render content correctly 1`] = `
<div
  class="monthly whiteframe md-whiteframe-3dp"
>
  <h2>
    Monthly Income Distrobution
  </h2>
   
  <div
    class="monthly-content"
  >
    <div
      class="month"
    >
      <span>
        MAR: $15
      </span>
    </div>
     
    <div
      class="month"
    >
      <span>
        APR: $15
      </span>
    </div>
     
    <div>
      <span>
        MAY: $15
      </span>
    </div>
     
    <div>
      <span>
        JUN: $15
      </span>
    </div>
     
    <div>
      <span>
        JUL: $15
      </span>
    </div>
     
    <div>
      <span>
        AUG: $15
      </span>
    </div>
     
    <div>
      <span>
        SEP: $15
      </span>
    </div>
     
    <div>
      <span>
        OCT: $15
      </span>
    </div>
     
    <div>
      <span>
        NOV: $15
      </span>
    </div>
     
    <div>
      <span>
        DEC: $15
      </span>
    </div>
     
    <div>
      <span>
        JAN: $15
      </span>
    </div>
     
    <div>
      <span>
        FEB: $15
      </span>
    </div>
  </div>
</div>
`;
